components/Chat/Chat.tsx

```diff
@@ -1,21 +1,12 @@
-import { useChatProvider } from "@/providers/ChatProvider";
+import { ScrollTo } from "react-scroll-to";
 import ChatInput from "./ChatInput";
 import Messages from "./Messages";
-import Suggestions from "./Suggestions";
 
 const Chat = () => {
```

Comment:
Utilize useEffect with a dependency on the messages array to auto-scroll to the latest message when messages changes. This will ensure that every new message triggers an automatic scroll.

const Chat = () => {
  const { messages, input, handleInputChange, handleSubmit } = useChatProvider();
  const messagesEndRef = useRef(null);

  // Auto-scroll to the latest message when the `messages` array changes
  useEffect(() => {
    if (messagesEndRef.current) {
      messagesEndRef.current.scrollIntoView({ behavior: "smooth" });
    }
  }, [messages]);